proxy/src/registry/transaction.rs

```diff
@@ -4,13 +4,43 @@ use async_trait::async_trait;
 use hex::ToHex;
 use kv::Codec as _;
 use serde::{Deserialize, Serialize};
-use std::time::SystemTime;
+use std::time::{self, Duration, SystemTime};
 
 use radicle_registry_client as protocol;
 
 use crate::error;
 use crate::registry;
 
+/// Amount of blocks we assume to have been mined before a transaction is
+/// considered to have settled.
+const SETTLEMENT_OFFSET: u32 = 6;
```

Comment:
More commonly known as "minimum confirmations". Also note that you're going to be off by one if you treat this as an offset, because when the tx is applied, that counts as the 1st confirmation, so you only need 5 more after that.